### PR TITLE
feat(chat): reconnect chat after page reload or when going back

### PR DESCRIPTION
**Pull request by jaroslav-kubicek**
- more analytics added
- cypress tests added
- reconnect after page reload

<br/><br/><br/><url>LiveURL: https://smartfaq-reconnect-chat-after-reload.surge.sh</url>

### CODE REVIEW

**Comment by jaroslav-kubicek** [comment]
finished, cc @jalbertsr @kantord & @vborszeki

File: src/relay/GuaranteeNeededResolver.js [hunk 1]
```diff
@@ -107,6 +108,12 @@ class GuaranteeNeededResolver extends React.Component<Props> {
     );
 
     if (showGuaranteeChat !== this.props.showGuaranteeChat) {
+      if (showGuaranteeChat) {
+        simpleTracker('smartFAQBookingOverview', {
+          action: 'chatEnabled',
```
Comment (jalbertsr) [inline src/relay/GuaranteeNeededResolver.js]: can you add this tracker to the README.md https://github.com/kiwicom/smart-faq#analytics so we have an updated list of all trackers running.

Comment (jaroslav-kubicek) [reply src/relay/GuaranteeNeededResolver.js]: added

File: src/StaticFAQ/ArticleDetail/GuaranteeChat.js [hunk 2]
```diff
@@ -31,74 +30,36 @@ class GuaranteeChat extends React.Component<Props, State> {
...
-
-    const body = document && document.body;
-
-    if (!body) {
-      return;
-    }
-
-    if (document.getElementById('purecloud-webchat-js')) {
+  async componentDidMount() {
+    if (typeof window !== 'object') {
```
Comment (kantord) [inline src/StaticFAQ/ArticleDetail/GuaranteeChat.js]: What is this for?

**Comment by kantord** [comment]
For some reason the chat doesn't load on surge.

>  **Error: Unexpected: webchat not initialized.** GuaranteeChat.js:51:12

2018-09-20-123501_3839x2025_scrot

**Comment by jalbertsr** [comment]
> For some reason the chat doesn't load on surge.

Yes, I received the sentry alert https://sentry.skypicker.com/kiwicom/smartfaqstaging/issues/3030027/

**Comment by jaroslav-kubicek** [comment]
> For some reason the chat doesn't load on surge.

Thanks for noticing @jalbertsr and @kantord , looks like sometimes it may take a moment to initialize chat instance and you may click on it before it's ready, therefore I propose to disable button until it's ready.

**Comment by kantord** [comment]
Now something weird is happening: it takes about 10 seconds for the button to become enabled.

**Comment by kantord** [comment]
If I reload with F5, the correct article is not loaded and chat is not forced. If I manually force it again and navigate back, then the chat is automatically loaded, however, I don't see my old message.

2018-09-20-150706_2513x1952_scrot

**Comment by jalbertsr** [comment]
same for me, I only get a blank chat, but no the old messages

**Comment by vborszeki** [comment]
the same here too

**Comment by jaroslav-kubicek** [comment]
> Now something weird is happening: it takes about 10 seconds for the button to become enabled.

yeah, it depends on request to PC and sometimes it takes a significant amount of time 🙃 

thanks for debugging, I tried to improve debugging by keeping `forceChat` value between reloads + fixed prompt. Try it one more time.

Speaking about old messages, it seems they are loaded only when chat was responded by an agent. it's limitation of PC

**Comment by jaroslav-kubicek** [comment]
is it better now? 🙂 we need to resolve this asap pls. @kantord @jalbertsr @vborszeki

**Comment by kantord** [comment]
Somehow, when I click the force guarantee chat button, nothing happens. It stays unchecked.

**Comment by jalbertsr** [comment]
> Somehow, when I click the force guarantee chat button, nothing happens. It stays unchecked.

same here I think `window.GuaranteeChatForce` stays false all the time, session cookie takes value 1 always.

**Comment by jaroslav-kubicek** [comment]
fixeeeeed 🙏 @jalbertsr & @kantord

**Pull request merged**